packages/core-extensions/src/moonbase/webpackModules/stores.ts

```diff
@@ -437,12 +437,19 @@ class MoonbaseSettingsStore extends Store<any> {
         }
       }
 
-      const initConfig = typeof initState === "boolean" ? {} : initState?.config ?? {};
-      const newConfig = typeof newState === "boolean" ? {} : newState?.config ?? {};
+      const initConfig = typeof initState === "boolean" ? {} : { ...initState?.config };
+      const newConfig = typeof newState === "boolean" ? {} : { ...newState?.config };
 
       const def = ext.manifest.settings;
       if (!def) continue;
 
+      for (const key in def) {
+        const defaultValue = def[key].default;
+
+        if (initConfig[key] == null) Object.defineProperty(initConfig, key, { value: defaultValue, enumerable: true });
+        if (newConfig[key] == null) Object.defineProperty(newConfig, key, { value: defaultValue, enumerable: true });
```

Comment:
Consider making these nullish coalescing assignments, that would be a bit more readable.

Reply:
I see, that is indeed an issue for `__proto__` only, however nowhere is this currently handled and properly using objects with null prototypes everywhere would better fit in a separate PR.